src/main/scala/tilelink/Parameters.scala

```diff
@@ -366,6 +395,35 @@ object TLSlaveParameters {
       alwaysGrantsT = alwaysGrantsT,
       fifoId        = fifoId)
   }
+
+  def v2(
+    nodePath:      Seq[BaseNode]                = Seq(),
+    resources:     Seq[Resource]                = Seq(),
+    setName:       String,
+    address:       Seq[AddressSet],
```

Comment:
I am a bit worried that we've made the non-default argument not the first argument.

The plus side: it will force people to write 'address = ' in the constructor. That's also the downside.

@hcook ?

Reply:
arguments with no default value should always go first IMO